plugin-server/src/utils/db/db.ts

```diff
@@ -194,12 +200,23 @@ export class DB {
 
     // Postgres
 
-    public postgresQuery<R extends QueryResultRow = any, I extends any[] = any[]>(
+    public async postgresQuery<R extends QueryResultRow = any, I extends any[] = any[]>(
         queryString: string,
         values: I | undefined,
         tag: string,
-        client?: PoolClient
-    ): Promise<QueryResult<R>> {
+        client?: PoolClient,
+        cacheResult = false,
+        cacheResultTtlSeconds = 15
+    ): Promise<PostgresQueryResult<R>> {
+        const cacheKey = `${POSTGRES_QUERY_CACHE_PREFIX}${tag}`
+
+        if (cacheResult) {
```

Comment:
Does this also avoid the [thundering herd problem](https://en.wikipedia.org/wiki/Thundering_herd_problem)

Not the end of the world if not, but would be nice to have this also, what do you think @tiina303 @guidoiaquinti ?

Reply:
Agreed, though I assume this is still an improvement. We were separately talking about moving that to the main thread, but across workers we still don't have a solution afaik.

Reply:
addressed this